kafka/cluster.go

```diff
@@ -149,6 +149,25 @@ func (module *Cluster) describeLogDirs() {
 
 	for _, broker := range brokers {
 		go func(b *sarama.Broker) {
+			connected, err := b.Connected()
+			if err != nil {
+				resCh <- response{
+					BrokerID: b.ID(),
+					Res:      nil,
+					Err:      err,
+				}
+			}
+			if !connected {
+				err = b.Open(module.client.Config())
+				if err != nil {
+					resCh <- response{
+						BrokerID: b.ID(),
+						Res:      nil,
+						Err:      err,
+					}
+				}
+			}
+
```

Comment:
```suggestion
			_ = b.Open(module.client.Config())
```

Reply:
This should be sufficient. The subsequent request to describe log dirs would then handle the error in case it is not connected. Since the error handling in this case is the same as you suggested this should be fine.

What do you think?

Reply:
my understanding is that `Open`[ returns `AlreadyConnected` ](https://godoc.org/github.com/Shopify/sarama#Broker.Open) so either handling that error explicitly or checking `Connected()` is required iiuc

Reply:
RIght and when it returns `AlreadyConnected` we are good to go. Checking `Connected()` would just block until we are connected which `Open()` does not ensure. Open is not blocking and only the next operation which actually requires a connection will await the connection then.

Reply:
right - ConfigurationError may also be thrown in Open.

i see your point in ignoring that error to simplify the code but felt it's more clear to handle this explicitly

Reply:
I don't think `ConfigurationError` can be thrown because we validate the sarama config here: https://github.com/cloudworkz/kafka-minion/blob/master/kafka/connection_helper.go#L92-L95 . I like being verbose as well, but I don't see any advantage of being that verbose here. You could leave a comment above the `b.Open()` line why this is enough.

Reply:
ok - agree - please see the latest changes